__unported__/sale_recurring_orders/recurring_orders.py

```diff
@@ -316,7 +317,7 @@ def generate_agreement_orders(self, cr, uid, agreement, startDate, endDate, cont
         agreement_order_obj = self.pool.get('sale.recurring_orders.agreement.order')
         for date in dates: 
             # Check if an order exists for that date
-            if not len(agreement_order_obj.search(cr, uid, [ ('date', '=', date), ('agreement_id', '=', agreement['id']) ])):
+            if not len(agreement_order_obj.search(cr, uid, [ ('date', '=', str(date)), ('agreement_id', '=', agreement['id']) ])):
```

Comment:
Si no lo hago con str me da este error: 
TypeError: object of type 'datetime.datetime' has no len()